controllers/gateway_controller.go

```diff
@@ -99,37 +96,63 @@ func (r *GatewayReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		return ctrl.Result{}, nil
 	}
 
+	gatewayReadyStatus, gatewayReadyStatusIsSet := isGatewayReady(gw)
+	oldGateway := gw.DeepCopy()
+	initGatewayStatus(gw)
+	factorizeStatus(gw, oldGateway)
+
 	log.Info("checking for Service for Gateway")
 	svc, err := r.getServiceForGateway(ctx, gw)
 	if err != nil {
 		return ctrl.Result{}, err
 	}
 	if svc == nil {
+		// if the ready status is not set, or the gateway is marked as ready, mark it as not ready
+		if !gatewayReadyStatusIsSet || gatewayReadyStatus {
+			return ctrl.Result{}, r.patchGatewayStatus(ctx, gw, oldGateway) // status patch will requeue gateway
```

Comment:
nit: I'm not sure we really need this `patchGatewayStatus` wrapper, we're trading one line with slightly more characters for one line with slightly less, but I feel readers will often feel compelled to read the implementation anyway to know what it's doing, so I kinda feel like we should just do the `Patch()` here :thinking:

Reply:
done :)